scripts/migrations/005_convert_compability_fileds_to_ternary.py

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/python
+# -----------------------------------------------------------------------------
+# Copyright (c) Bosch Software Innovations GmbH 2017.
+# Part of the SW360 Portal Project.
+#
+# All rights reserved.   This program and the accompanying materials
+# are made available under the terms of the Eclipse Public License v1.0
+# which accompanies this distribution, and is available at
+# http://www.eclipse.org/legal/epl-v10.html
+#
+# This is a manual database migration script. It is assumed that a
+# dedicated framework for automatic migration will be written in the
+# future. When that happens, this script should be refactored to conform
+# to the framework's prerequisites to be run by the framework. For
+# example, server address and db name should be parametrized, the code
+# reorganized into a single class or function, etc.
+#
+# -----------------------------------------------------------------------------
+
+import couchdb
+
+COUCHSERVER = "http://localhost:5984/"
+DBNAME = 'sw360db'
+
+couch=couchdb.Server(COUCHSERVER)
+db = couch[DBNAME]
+
+
+licenses_by_id_fun = '''function(doc){
+    if (doc.type=="license"){
+        emit(doc._id, doc)
+    }
+}'''
+
+licenses = db.query(licenses_by_id_fun)
+
+def convertToTernary(compabilityBool):
+    if isinstance(compabilityBool, bool):
+        if compabilityBool:
+            return "YES"
+        else:
+            return "UNDEFINED"
+    return compabilityBool
+
+print 'Converting boolean compability flags to ternary.'
+for license_row in licenses:
+    license = license_row.value
+    for field in ['GPLv2Compat','GPLv3Compat']:
+        if field in license:
+            license[field] = convertToTernary(license[field])
+        else:
+            license[filed] = "UNDEFINED"
+    if 'issetBitfield' in license:
```

Comment:
In Thrift, if you have some bool value, it is by default `False` and you do not know whether it was actually set. This solves `issetBitfield`.

If `issetBitfiled==0010100...₂` on a binary level, this means that the third and the fifth basic value is set by intent.

Since we remove the last basic value we also remove `issetBitfield` (otherwise the deserializer has problems).